src/libvirtApi/storagePool.js

```diff
@@ -43,19 +43,19 @@ export function storagePoolCreate({
     autostart,
 }) {
     const poolXmlDesc = getPoolXML({ name, type, source, target });
-    let storagePoolPath;
 
-    return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolDefineXML', [poolXmlDesc, 0], { timeout, type: 'su' })
-            .then(poolPath => {
-                storagePoolPath = poolPath[0];
-                const args = ['org.libvirt.StoragePool', 'Autostart', cockpit.variant('b', autostart)];
-
-                return call(connectionName, storagePoolPath, 'org.freedesktop.DBus.Properties', 'Set', args, { timeout, type: 'ssv' });
-            }, exc => {
-                if (storagePoolPath)
-                    storagePoolUndefine({ connectionName, objPath: storagePoolPath });
-                return Promise.reject(exc);
-            });
+    let storagePoolPath;
+    try {
+        [storagePoolPath] = await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'StoragePoolDefineXML',
+                                       [poolXmlDesc, 0], { timeout, type: 'su' });
+        const args = ['org.libvirt.StoragePool', 'Autostart', cockpit.variant('b', autostart)];
+
+        await call(connectionName, storagePoolPath, 'org.freedesktop.DBus.Properties', 'Set', args, { timeout, type: 'ssv' });
+    } catch (ex) {
+        if (storagePoolPath)
+            storagePoolUndefine({ connectionName, objPath: storagePoolPath });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1400-20240125-085010-77feb7f2-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)